trinity/sync/common/chain.py

```diff
@@ -99,19 +118,18 @@ def register_peer(self, peer: BasePeer) -> None:
         self.run_task(self._handle_msg_loop())
         with self.subscribe(self.peer_pool):
             while not self.cancel_token.triggered:
-                peer_or_finished: Any = await self.wait_first(
-                    self._sync_requests.get(),
-                    self._sync_complete.wait()
-                )
-
-                # In the case of a fast sync, we return once the sync is completed, and our caller
-                # must then run the StateDownloader.
-                if self._sync_complete.is_set():
+                try:
+                    print('waiting for peer to sync')
+                    peer = await self.cancel_token.cancellable_wait(self._sync_requests.get())
```

Comment:
Is there a reason not to use `self.wait`?

Reply:
:+1: There used to be, but it's gone.